bee-api/bee-rest-api/src/endpoints/auth.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use auth_helper::jwt::{Claims, JsonWebToken, TokenData};
+use axum::{
+    async_trait,
+    extract::{Extension, FromRequest, OriginalUri, RequestParts, TypedHeader},
+    headers::{authorization::Bearer, Authorization},
+    http::Uri,
+};
+use lazy_static::lazy_static;
+use regex::RegexSet;
+use serde::{Deserialize, Serialize};
+
+use crate::endpoints::{config::route_to_regex, error::ApiError, storage::StorageBackend, ApiArgsFullNode};
+
+pub const API_AUDIENCE_CLAIM: &str = "api";
+pub const DASHBOARD_AUDIENCE_CLAIM: &str = "dashboard";
+const API_JWT_HINT: &str = "\"aud\":\"api\"";
+#[cfg(feature = "dashboard")]
+const DASHBOARD_JWT_HINT: &str = "\"aud\":\"dashboard\"";
+
+lazy_static! {
+    static ref DASHBOARD_ROUTES: RegexSet = {
+        let routes = vec![
+            "/api/v2/info",
+            "/api/v2/messages/*",
+            "/api/v2/outputs/*",
+            "/api/v2/addresses/*",
+            "/api/v2/milestones/*",
+            "/api/v2/peers*",
+        ];
+        // Panic: unwrapping is fine because all strings in `routes` can be turned into valid regular expressions.
+        RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<_>>()).unwrap()
+    };
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Auth<S> {
+    phantom: PhantomData<S>,
+}
+
+#[async_trait]
+impl<B, S> FromRequest<B> for Auth<S>
+where
+    B: Send + Sync,
+    S: StorageBackend,
+{
+    type Rejection = ApiError;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        let OriginalUri(uri) = OriginalUri::from_request(req).await.map_err(|_| ApiError::Forbidden)?;
+
+        let Extension(args) = Extension::<ApiArgsFullNode<S>>::from_request(req)
+            .await
+            .map_err(|_| ApiError::Forbidden)?;
+
+        // Check if the requested endpoint is open for public use.
+        if args.rest_api_config.public_routes().is_match(&uri.to_string()) {
+            return Ok(Auth { phantom: PhantomData });
+        }
+
+        // Extract the token from the authorization header
```

Comment:
```suggestion
        // Extract the token from the authorization header.
```
Can't help myself :trollface: